courses/management/commands/manage_program_certificates.py

```diff
@@ -0,0 +1,75 @@
+"""
+Management command to sync grades and certificates for a course run
+"""
+from django.core.management.base import BaseCommand, CommandError
+from django.db.models import Q
+
+from courses.models import ProgramEnrollment
+from courses.utils import generate_program_certificate
+from users.api import fetch_user
+
+
+class Command(BaseCommand):
+    """
+    Command to create program certificates for users.
+    """
+
+    help = "Create program certifificates, specific or all, for user(s)."
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--user",
+            type=str,
+            help="The id, email or username of the enrolled User",
+            required=False,
+        )
+        parser.add_argument(
+            "--readable_id",
+            type=str,
+            help="The 'readable_id' value for a Program",
+            required=True,
+        )
+        super().add_arguments(parser)
+
+    def handle(
+        self, *args, **options
+    ):  # pylint: disable=too-many-locals,too-many-branches
+        """Handle command execution"""
+        program = options.get("readable_id")
+        if not program:
+            raise CommandError("Please provide a valid program readable_id.")
+
+        user = options.get("user") and fetch_user(options["user"])
+        base_query = (
+            Q(program__readable_id=program, user=user)
+            if user
+            else Q(program__readable_id=program)
+        )
+
+        enrollments = ProgramEnrollment.objects.filter(base_query)
+        if not enrollments:
+            raise CommandError(
+                f"Could not find program enrollment(s) with readable_id={program}"
+            )
```

Comment:
This might not be a viable solution for following reason(s):

- A program enrollment might not always exist for user who is enrolled in a course for the program, We try to create program enrollment during a program certificate generation, Please take a look at https://github.com/mitodl/mitxpro/blob/master/courses/utils.py#L147

So a viable approach here would be:

1. Acquire the courses associated with the program since we will always have a program as input to the command
2. Get the users based on course enrollments in the courses acquired in point-1.
3. Now that you have a program and list of users, Run `generate_program_certificate` which would handle both, Certificate generation and program enrollment.